src/Seq.App.Opsgenie/OpsgenieApp.cs

```diff
@@ -71,14 +145,63 @@ public async Task OnAsync(Event<LogEventData> evt)
         {
             if (evt == null) throw new ArgumentNullException(nameof(evt));
 
-            var alert = new OpsgenieAlert(
-                _generateMessage.Render(evt),
-                evt.Id,
-                _generateDescription.Render(evt),
-                Host.BaseUri,
-                _tags);
+            List<string> tagList = _tags.ToList<string>();
+
+            if (_includeTags && evt.Data.Properties.ContainsKey(_includeTagProperty))
+                {
+                    var property = evt.Data.Properties[_includeTagProperty];
+                    if (property.GetType().IsArray)
+                        foreach (object p in (object[])property)
+                            if (!string.IsNullOrEmpty((string)p) && !tagList.Contains((string)p, StringComparer.CurrentCultureIgnoreCase))
+                                tagList.Add(((string)p).Trim());
+                            else if (!string.IsNullOrEmpty((string)p) && !tagList.Contains((string)p, StringComparer.CurrentCultureIgnoreCase))
+                                tagList.AddRange(((string)property).Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(t => t.Trim()));
+                }
 
-            await ApiClient.CreateAsync(alert);
+            try
+            {
+                
+                //Log our intent to alert OpsGenie with details that could be re-fired to another app if needed
+                Log.Debug("Send Alert to OpsGenie: Message {Message}, Description {Description}, Priority {Priority}, Responders {Responders}, Tags {Tags}", _generateMessage.Render(evt), _generateDescription.Render(evt),
+                    _priority, responder, tagList.ToArray());
+
+                //Logging the API call helps with debugging "why" an alert did not fire or was rejected by OpsGenie API
+                Log.Debug("OpsGenie API call: {JsonCall}", JsonSerializer.Serialize(new OpsgenieAlert(
+                        _generateMessage.Render(evt),
+                        evt.Id,
+                        _generateDescription.Render(evt),
+                        _priority,
+                        _responders,
+                        Host.BaseUri,
+                        tagList.ToArray()), new JsonSerializerOptions()
```

Comment:
Perhaps we could add a static `Serialize()` method on `OpsGenieApiClient` and avoid some of the duplicated serializer setup between these two classes?

Reply:
Makes sense. I was going in a straight line for implementation and hadn't rounded back on possible improvements.